cmake/GoogleCloudCppLibrary.cmake

```diff
@@ -41,7 +41,7 @@ function (google_cloud_cpp_add_ga_grpc_library library display_name)
         list(APPEND source_globs "${dir}*.h" "${dir}*.cc" "${dir}internal/*")
         list(APPEND mocks_globs "${dir}mocks/*.h")
         list(APPEND DOXYGEN_EXCLUDE_SYMBOLS "${library}_${ns}internal")
-        if (NOT dir STREQUAL "")
+        if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/${dir}samples")
```

Comment:
FWIW, `bigquery`, `iam` and `pubsublite` also have this directory.

Reply:
Ack. Those are outside of the scope of #8022.